robottelo/host_helpers/satellite_mixins.py

```diff
@@ -107,6 +110,33 @@ def md5_by_url(self, url):
             f'wget -qO - {url} | tee {filename} | md5sum | awk \'{{print $1}}\''
         ).stdout
 
+    def upload_manifest(self, org_id, manifest=None, interface='API', timeout=None):
+        """Upload a manifest using the requested interface.
+
+        :type org_id: int
+        :type manifest: Manifester object or None
+        :type interface: str
+        :type timeout: int
+
+        :returns: the manifest upload result
+
+        """
+        if manifest is None:
+            manifest = clone().content
+        if timeout is None:
+            # Set the timeout to 1500 seconds to align with the API timeout.
+            timeout = 1500000
+        if interface == 'CLI':
+            self.put(manifest, manifest.filename)
+            result = self.cli.Subscription.upload(
+                {'file': manifest.filename, 'organization-id': org_id}, timeout=timeout
```

Comment:
I mentioned this in a comment further down and just realized that it's a consideration for failing over to cloned manifests as well. We'll need to account for the differences in attributes between the response objects returned by manifester and manifest objects in robottelo. A cloned manifest should use what's written here, but a manifester manifest will need to do the following:
```
self.put(f'{manifest.path}', f'{manifest.name}')
result = self.cli.Subscription.upload(
    {'file': manifest.name, 'organization-id': org_id}, timeout=timeout
)
```
The f-strings are necessary in the `put` command because those to manifester object attributes are Path objects and put expects to be passed strings.